cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -98,6 +106,53 @@ func (w *Watcher) check() {
 func (w *Watcher) RecordUsage(e entry) {
 	logging.Debug("Recording usage of %s (%d)", e.Exec, e.PID)
 	w.an.Event(anaConst.CatRuntimeUsage, anaConst.ActRuntimeHeartbeat, e.Dims)
+	w.UpdateLastUsed(e)
+}
+
+func (w *Watcher) UpdateLastUsed(e entry) {
+	if stateExec, err := installation.StateExec(); err == nil {
+		if isStateExec, err := fileutils.PathsEqual(e.Exec, stateExec); err == nil && isStateExec {
+			return // cannot infer which runtime is being used
+		} else if err != nil {
+			multilog.Error("Could not determine if %s is the State Tool executable: %s", e.Exec, errs.JoinMessage(err))
+		}
+	} else {
+		multilog.Error("Could not determine the state tool executable: %s", errs.JoinMessage(err))
+	}
```

Comment:
This still needs work to try and infer runtime from e.g. `state exec`.

Reply:
This might be tricky because all I have to work with is e.Dims, which has a namespace, but there might be multiple checkouts for that namespace.